spec/roo_on_rails/sidekiq/metrics_worker_spec.rb

```diff
@@ -5,148 +5,164 @@
 
   describe '#perform' do
     let(:perform) { described_class.new.perform }
+
     before do
-      allow(statsd).to receive(:batch).and_call_original
-      allow(statsd).to receive(:gauge)
+      allow(statsd).to receive(:gauge).with(any_args)
+      allow(Sidekiq::Queue).to receive(:all).and_return(sidekiq_queues) if defined?(sidekiq_queues)
     end
 
     after { reset_queues }
 
     it 'should send all metrics in a batch' do
+      expect(statsd).to receive(:batch).with(no_args).once
```

Comment:
rspec has 2 method-call expectations. `.to receive()` and `.to have_received()`. the former requires the call to be before the action is called. this way if we received some unexpected call, an error is raised. the former works but doesn't check for any 'wrong' calls.